lib/improver/tests/nowcasting/utilities/test_ApplyOrographicEnhancement.py

```diff
@@ -0,0 +1,584 @@
+
+    """Test the _select_orographic_enhancement method."""
+
+    def setUp(self):
+        """Set up cubes for testing."""
+        self.precip_cubes = set_up_precipitation_rate_cube()
+        self.oe_cube = set_up_orographic_enhancement_cube().concatenate_cube()
+        self.first_slice = self.oe_cube[:, 0, :, :]
+        self.second_slice = self.oe_cube[:, 1, :, :]
+
+    def test_basic(self):
+        """Test extracting a time coordinate from the orographic enhancement
+        cube."""
+        plugin = ApplyOrographicEnhancement("add")
+        result = plugin._select_orographic_enhancement_cube(
+            self.precip_cubes[0], self.oe_cube)
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.metadata, self.first_slice.metadata)
+        self.assertEqual(result, self.first_slice)
+        self.assertEqual(result.coord("time"), self.first_slice.coord("time"))
+
+    def test_alternative_time_quarter_past(self):
+        """Test extracting a time coordinate from the orographic enhancement
+        cube at quarter past an hour."""
+        plugin = ApplyOrographicEnhancement("add")
+        self.precip_cubes[0].coord("time").points = 412227.25
+        result = plugin._select_orographic_enhancement_cube(
+            self.precip_cubes[0], self.oe_cube)
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.metadata, self.first_slice.metadata)
+        self.assertEqual(result, self.first_slice)
+        self.assertEqual(result.coord("time"), self.first_slice.coord("time"))
+
+    def test_alternative_time_half_past(self):
+        """Test extracting a time coordinate from the orographic enhancement
+        cube at half past an hour. Note that the time point will round down
+        at the midpoint between 412227. and 412228."""
+        plugin = ApplyOrographicEnhancement("add")
+        self.precip_cubes[0].coord("time").points = 412227.5
+        result = plugin._select_orographic_enhancement_cube(
+            self.precip_cubes[0], self.oe_cube)
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.metadata, self.first_slice.metadata)
+        self.assertEqual(result, self.first_slice)
+        self.assertEqual(result.coord("time"), self.first_slice.coord("time"))
+
+    def test_alternative_time_quarter_to(self):
+        """Test extracting a time coordinate from the orographic enhancement
+        cube at quarter to an hour."""
+        plugin = ApplyOrographicEnhancement("add")
+        self.precip_cubes[0].coord("time").points = 412227.75
+        result = plugin._select_orographic_enhancement_cube(
+            self.precip_cubes[0], self.oe_cube)
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.metadata, self.second_slice.metadata)
+        self.assertEqual(result, self.second_slice)
+        self.assertEqual(result.coord("time"), self.second_slice.coord("time"))
```

Comment:
Can this function also test for having multiple times on the precip_cube and for two times matching one target orographic_enhancement time?

Reply:
Although I've added a copy of the cube, I've also added a unit test to check that the orographic enhancement is not being modified, as intended.

Reply:
@MoseleyS I've added an additional unit test, as suggested, to test for where multiple input precipitation fields match a single orographic enhancement cube.